sentry-android-navigation/src/test/java/io/sentry/android/navigation/SentryNavigationListenerTest.kt

```diff
@@ -106,14 +163,181 @@ class SentryNavigationListenerTest {
+
+        sut.onDestinationChanged(fixture.navController, fixture.destination, null)
+
+        verify(fixture.hub, never()).startTransaction(
+            any(),
+            any(),
+            any(),
+            anyOrNull(),
+            any<Boolean>()
+        )
+    }
+
+    @Test
+    fun `onDestinationChanged does not start tracing when tracesSampleRate is not set`() {
+        val sut = fixture.getSut(enableTracing = true, tracesSampleRate = null)
+
+        sut.onDestinationChanged(fixture.navController, fixture.destination, null)
+
+        verify(fixture.hub, never()).startTransaction(
+            any(),
+            any(),
+            any(),
+            anyOrNull(),
+            any<Boolean>()
         )
     }
+
+    @Test
+    fun `onDestinationChanged does not start tracing when navigating between activities`() {
+        val sut = fixture.getSut()
+        whenever(fixture.destination.navigatorName).thenReturn("activity")
+
+        sut.onDestinationChanged(fixture.navController, fixture.destination, null)
+
+        verify(fixture.hub, never()).startTransaction(
+            any(),
+            any(),
+            any(),
+            anyOrNull(),
+            any<Boolean>()
+        )
+    }
+
+    @Test
+    fun `onDestinationChanged does not start tracing when route and id are not available`() {
+        val sut = fixture.getSut(toRoute = null, hasViewIdInRes = false)
+
+        sut.onDestinationChanged(fixture.navController, fixture.destination, null)
+
+        verify(fixture.hub, never()).startTransaction(
+            any(),
+            any(),
+            any(),
+            anyOrNull(),
+            any<Boolean>()
+        )
+    }
+
+    @Test
+    fun `onDestinationChanged starts tracing with the route name as transaction name`() {
+        val sut = fixture.getSut()
+
+        sut.onDestinationChanged(fixture.navController, fixture.destination, null)
+
+        verify(fixture.hub).startTransaction(
+            check { assertEquals("/route", it) },
+            check { assertEquals(SentryNavigationListener.NAVIGATION_OP, it) },
+            any(), anyOrNull(), any<Boolean>()
+        )
+    }
+
+    @Test
+    fun `onDestinationChanged strips out route parameters from transaction name`() {
+        val sut = fixture.getSut(toRoute = "github/{user_id}?per_page={per_page}")
+
+        sut.onDestinationChanged(fixture.navController, fixture.destination, null)
+
+        verify(fixture.hub).startTransaction(
+            check { assertEquals("/github", it) },
+            any(), any(), anyOrNull(), any<Boolean>()
+        )
+    }
+
+    @Test
+    fun `onDestinationChanged starts tracing with destination id if route is not available`() {
+        val sut = fixture.getSut(toRoute = null, hasViewIdInRes = true)
+
+        sut.onDestinationChanged(fixture.navController, fixture.destination, null)
+
+        verify(fixture.hub).startTransaction(
+            check { assertEquals("/id", it) },
```

Comment:
Should we use something a little more unique instead of `id` as destination ID, so it's easier to find where this is set in the test?

Reply:
mm, I'm not sure I got the problem - what is easier to find?

Reply:
```suggestion
            check { assertEquals("/destination-id-1", it) },
```

Maybe like this. This isn't really important though.

Reply:
oh yeah we can do that